llvm/test/CodeGen/DirectX/comput_ids.ll

```diff
@@ -18,7 +18,7 @@ entry:
 ; Function Attrs: noinline nounwind optnone
 define i32 @test_group_id(i32 %a) #0 {
 entry:
-; CHECK: call i32 @dx.op.groupId.i32(i32 94, i32 %{{.*}})
+; CHECK: call i32 @dx.op.unary.i32(i32 94, i32 %{{.*}})
```

Comment:
This will be illegal DXIL since function name mismatch with dxil opcode.

Reply:
Is there a specific format the name of DXIL operation call is expected to follow? Can you please point me to such a specification? Thanks!

Reply:
I'm not aware of a source of truth for what names are used other than the logic that generates the dxil op code in dxc: https://github.com/microsoft/DirectXShaderCompiler/blob/main/utils/hct/hctdb.py#L2613

Unfortunately there isn't much said about the actual function names in the DXIL specification, other than to mention that `dx.op.unary` exists: https://github.com/microsoft/DirectXShaderCompiler/blob/main/docs/DXIL.rst#L2107

Reply:
It is not in any doc I guess :(

The code is in https://github.com/microsoft/DirectXShaderCompiler/blob/main/lib/DXIL/DxilOperations.cpp#L2693
ConstructOverloadName is used when create a DXIL op function.

Reply:
> The code is in https://github.com/microsoft/DirectXShaderCompiler/blob/main/lib/DXIL/DxilOperations.cpp#L2693 ConstructOverloadName is used when create a DXIL op function.

Thanks!

> It is not in any doc I guess :(

If it is not documented (and consequently, as it appears to me, would not provide a standard to follow for any consumer of the generated DXIL code), which runtime/execution component considers `dx.op.groupId.i32(i32 94, ...)` as legal  but `dx.op.unary.i32(i32 94, ...)` as illegal?

Reply:
The validator should catch it.

Reply:
> The validator should catch it. The normal way to consume is thru DXILOperation https://github.com/microsoft/DirectXShaderCompiler/blob/main/include/dxc/DXIL/DxilOperations.h#L36 which will do the check.

Thanks for your help. I verified that the validator does catch it, as you said.

Will make changes to address this.

Reply:
Changes pushed.